src/dotty/tools/dotc/transform/Space.scala

```diff
@@ -14,35 +14,41 @@ import core.Decorators._
   */
 
 
-// space definition
+/** space definition */
 sealed trait Space
-object Empty extends Space
+case object Empty extends Space
 case class Typ(t: Type) extends Space
 case class Kon(t: Type, params: List[Space]) extends Space
 case class Or(spaces: List[Space]) extends Space
 
-// abstract space logic
+/** abstract space logic */
 trait SpaceLogic {
-  def isSubType(t1: Type, t2: Type): Boolean
-  def isEqualType(t1: Type, t2: Type): Boolean
-  def isCaseClass(t: Type): Boolean
+  /** is `tp1` a subtype of `tp2` */
```

Comment:
Sorry for being pedantic, but please start comments with a capital letter :). And put `?` at the end if it's a question.